plugin.video.viaplay/addon.py

```diff
@@ -3,601 +3,415 @@
 
+def list_next_page(url, action):
+    title = helper.language(30018)
+    params = {
+        'action': action,
+        'url': url
+    }
+    helper.add_item(title, params)
 
-def return_art(product, content):
-    """Return the available art in a xbmcgui.setArt friendly dict."""
-    try:
-        boxart = product['content']['images']['boxart']['url'].split('.jpg')[0] + '.jpg'
-    except KeyError:
-        boxart = None
-    try:
-        hero169 = product['content']['images']['hero169']['template'].split('.jpg')[0] + '.jpg'
-    except KeyError:
-        hero169 = None
-    try:
-        coverart23 = product['content']['images']['coverart23']['template'].split('.jpg')[0] + '.jpg'
-    except KeyError:
-        coverart23 = None
-    try:
-        coverart169 = product['content']['images']['coverart23']['template'].split('.jpg')[0] + '.jpg'
-    except KeyError:
-        coverart169 = None
-    try:
-        landscape = product['content']['images']['landscape']['url'].split('.jpg')[0] + '.jpg'
-    except KeyError:
-        landscape = None
 
-    if content == 'episode' or content == 'sport':
-        thumbnail = landscape
-    else:
-        thumbnail = boxart
-    fanart = hero169
-    banner = landscape
-    cover = coverart23
-    poster = boxart
+def list_products(url, filter_event=False, search_query=None):
+    if filter_event:
+        filter_event = filter_event.split(', ')
 
-    art = {
-        'thumb': thumbnail,
-        'fanart': fanart,
-        'banner': banner,
-        'cover': cover,
-        'poster': poster
+    products_dict = helper.vp.get_products(url, filter_event=filter_event, search_query=search_query)
+    for product in products_dict['products']:
+        if product['type'] == 'series':
+            add_series(product)
+        elif product['type'] == 'episode':
+            add_episode(product)
+        elif product['type'] == 'movie':
+            add_movie(product)
+        elif product['type'] == 'sport':
+            add_sports_event(product)
+        elif product['type'] == 'tvEvent':
+            add_tv_event(product)
+        else:
+            helper.log('product type: {0} is not (yet) supported.'.format(product['type']))
+            return False
+
+    if products_dict['next_page']:
+        list_next_page(products_dict['next_page'], 'list_products')
+    helper.eod()
+
+
+def add_movie(movie):
+    params = {}
+    if movie['system'].get('guid'):
+        params['guid'] = movie['system']['guid']
+    else:
+        params['url'] = movie['_links']['self']['href']
+    params['action'] = 'play'
+
+    details = movie['content']
+
+    movie_info = {
+        'mediatype': 'movie',
+        'title': details['title'],
+        'plot': details.get('synopsis'),
+        'genre': ', '.join([x['title'] for x in movie['_links']['viaplay:genres']]),
+        'year': details['production'].get('year'),
+        'duration': int(details['duration'].get('milliseconds')) / 1000 if 'duration' in details else None,
```

Comment:
In python3 dividing 2 integers will result in a float. You can use the `//` operator to ensure the result is always an integer. Not something you should worry right now, just pointing issues you'll have when moving the plugin to python3.

Reply:
Definitely noted. Thanks! :-)